### PR TITLE
C#: Fix isEffectively* visibility predicates

### PR DESCRIPTION
**Pull request by tamasvajk**
Extracted from https://github.com/github/codeql/pull/5664.

[Differences job](https://jenkins.internal.semmle.com/job/Changes/job/CSharp-Differences/1130/)

### CODE REVIEW

File: csharp/ql/src/semmle/code/csharp/Member.qll [hunk 1]
```diff
@@ -91,27 +91,59 @@ class Modifiable extends Declaration, @modifiable {
...
+
   /**
-   * Holds if this declaration is effectively `private` (either directly or
-   * because one of the enclosing types is `private`).
+   * Holds if this declaration is effectively `private`. A declaration is considered
+   * effectively `private` if it can only be referenced from
+   * - the declaring and its nested types, similarly to `private` declarations, and
+   * - the enclosing types.
+   *
+   * Note that explicit interface implementation are also considered effectively
```
Comment (hvitved) [inline csharp/ql/src/semmle/code/csharp/Member.qll:107]: implementations

File: csharp/ql/src/semmle/code/csharp/Member.qll [hunk 2]
```diff
@@ -91,27 +91,59 @@ class Modifiable extends Declaration, @modifiable {
...
   }
 
   /**
-   * Holds if this declaration is effectively `internal` (either directly or
-   * because one of the enclosing types is `internal`).
+   * Holds if this declaration is effectively `internal`. A declaration is considered
+   * effectively `internal` if it can only be referenced from the declaring assembly.
+   *
+   * Note that friend assemblies declared in `InternalsVisibleToAttribute` are not
+   * considered. Explicit interface implementation are also considered effectively
```
Comment (hvitved) [inline csharp/ql/src/semmle/code/csharp/Member.qll:133]: implementations

File: csharp/ql/src/semmle/code/csharp/Member.qll [hunk 3]
```diff
@@ -154,6 +186,11 @@ class Virtualizable extends Member, @virtualizable {
     implementsExplicitInterface()
   }
 
+  override predicate isPrivate() {
+    Member.super.isPrivate() and
```
Comment (hvitved) [inline csharp/ql/src/semmle/code/csharp/Member.qll:190]: The `Member.` suffix is no longer needed.

**Review by hvitved — CHANGES_REQUESTED** [review]
LGTM. Some trivial comments.

**Review by hvitved — DISMISSED** [review]

**Comment by tamasvajk** [comment]
I also rebased this PR to fix the merge conflicts.

**Review by hvitved — APPROVED** [review]

**Pull request merged**